### PR TITLE
Add more logging for debugging purposes

### PR DESCRIPTION
**Pull request by t1m0thyj**
## Proposed changes

Adds a log message inside most `catch` statements, which should make it easier to debug problems. Thanks @zFernand0 for assisting with this. 🙂 

## Release Notes

Milestone: 2.5.0

Changelog: Improved logging when Zowe Explorer encounters an error.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [ ] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [ ] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

**Review by JillieBeanSim — COMMENTED** [review]
@t1m0thyj with added logging should there be a CHANGELOG update for users?

**Review by JillieBeanSim — DISMISSED** [review]
LGTM! thanks @t1m0thyj for the addition of more logging

**Review by zFernand0 — DISMISSED** [review]

**Review by zFernand0 — DISMISSED** [review]

**Review by zFernand0 — DISMISSED** [review]

**Review by zFernand0 — DISMISSED** [review]
TL;DR: LGTM! 😋 

I believe we agreed on ignoring the `codecov/patch` for this particular PR since it touches so many different files that could use some better coverage 😋 
image

**Note:** I might be remembering wrong though 😅

**Review by zFernand0 — APPROVED** [review]

**Review by JillieBeanSim — APPROVED** [review]
